core-bundle/src/DependencyInjection/Filesystem/FilesystemConfiguration.php

```diff
@@ -88,7 +88,7 @@ public function mountAdapter(string $adapter, array $options, string $mountPath,
         $name ??= str_replace(['.', '/', '-'], '_', Container::underscore($mountPath));
         $adapterId = "contao.filesystem.adapter.$name";
 
-        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options))) {
+        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options, null))) {
```

Comment:
Other possibilities would be
```suggestion
        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options, Visibility::PUBLIC))) {
```
or:
```suggestion
        if (null !== ($adapterDefinition = $this->adapterDefinitionFactory->createDefinition($adapter, $options, Visibility::PRIVATE))) {
```

Togehter with a `use League\Flysystem\Visibility;`

Or adding a parameter to the method that gets passed on?

Maybe @m-vo can help here?

Reply:
Passing `null` seems to restore the previous defaults, as far as I can see. However this definitely needs a review.

Reply:
It seems that the new argument will be optional in the next release: https://github.com/thephpleague/flysystem-bundle/pull/156

Reply:
Ah nice. I guess we can close this PR then and revert https://github.com/contao/conflicts/pull/51

Reply:
So we don’t need to do anything, right? 🎉

Reply:
For the record: if things like this turn out to be a problem in the future, we could also remove the dependency to `league/flysystem-bundle` completely and handle things ourselves. Right now we are more or less just using the the `AdapterDefinitionFactory` which helps to turn the (rather specific) adapter config into services.